src/main/java/io/jenkins/plugins/pipelinegraphview/cards/items/GitHubBranchSourceRunDetailsItems.java

```diff
@@ -30,32 +31,24 @@ public static List<RunDetailsItem> getGitInformation(SCMRevisionAction scmRevisi
         String sourceRepo = head.getSourceRepo();
         String sourceBranch = head.getSourceBranch();
 
-        RunDetailsItem gitRepositoryItem = new RunDetailsItem.Builder()
-                .ionicon("logo-github")
-                .text(sourceOwner + "/" + sourceRepo)
-                .build();
-        runDetailsItems.add(gitRepositoryItem);
+        RunDetailsItem gitRepositoryItem = new RunDetailsItem.RunDetail(
+                new Ionicon("logo-github"), ItemContent.of(sourceOwner + "/" + sourceRepo));
 
-        RunDetailsItem gitBranchItem = new RunDetailsItem.Builder()
-                .ionicon("git-branch-outline")
-                .text(sourceBranch)
-                .build();
-        runDetailsItems.add(gitBranchItem);
+        RunDetailsItem gitBranchItem =
+                new RunDetailsItem.RunDetail(new Ionicon("git-branch-outline"), ItemContent.of(sourceBranch));
 
-        return runDetailsItems;
+        return List.of(gitRepositoryItem, gitBranchItem);
     }
 
     public static Optional<RunDetailsItem> getGitHubLink(WorkflowRun run) {
         GitHubLink gitHubLink = run.getParent().getAction(GitHubLink.class);
-        if (gitHubLink != null) {
-            ObjectMetadataAction action = run.getParent().getAction(ObjectMetadataAction.class);
-            RunDetailsItem build = new RunDetailsItem.Builder()
-                    .ionicon("git-pull-request-outline")
-                    .text(action.getObjectDisplayName())
-                    .href(gitHubLink.getUrl())
-                    .build();
-            return Optional.of(build);
+        if (gitHubLink == null) {
+            return Optional.empty();
         }
-        return Optional.empty();
+        ObjectMetadataAction action = run.getParent().getAction(ObjectMetadataAction.class);
+        return Optional.of(new RunDetailsItem.RunDetail(
+                new Ionicon("git-pull-request-outline"),
+                ItemContent.of(
+                        gitHubLink.getUrl(), requireNonNullElse(action.getObjectDisplayName(), gitHubLink.getUrl()))));
```

Comment:
what do we want to do here? The NonNull annotations highlighted that `action.getObjectDisplayName()` can return null

Reply:
if its null then I wouldn't show it